build: avoid run_text_test in teamcity-acceptance.sh

tbg opened the pull request:
I figured out how to do this properly. Let's get rid of run_text_test.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50691)
<!-- Reviewable:end -->

otan commented on build/teamcity-compose.sh:
```diff
@@ -22,9 +22,7 @@ run build/builder.sh mkrelease "$type" -Otarget testbuild PKG=./pkg/compose TAGS
-run cd ../..
+# NB: we're cheating go test into invoking our `compose.test` over the one it
+# builds itself.
+run_json_test go test -json -v -timeout 30m -exec ../../build/go-test-precompiled.sh ./compose.test ./pkg/compose
```
you may need -mod=vendor here and above.

tbg committed:
build: avoid run_text_test in teamcity-acceptance.sh

I figured out how to do this properly. Let's get rid of run_text_test.

Release note: None

otan reviewed (approved):
LGTM!

tbg commented:
I'm on vacation this week, I think this would pass if I invoked the test binary from `./pkg/acceptance` instead of repo root. Feel free to make that change and merge the PR, otherwise will get to it next wk.

otan commented:
Having trouble getting this to work with 1.14 (the previous error was just because it was on top of a flaky make lint) -- need to figure out how to get the GOROOT to work.
I'll give this a few more bashes to see if I can get it to work.

otan commented:
looks like a roachtest flake - it works!

bors r+

otan commented:
bors r+

On Wed, Jul 1, 2020 at 7:46 PM craig[bot] <notifications@github.com> wrote:

> Build failed
>
>    - GitHub CI (Cockroach)
>    <https://teamcity.cockroachdb.com/viewLog.html?buildId=2057829&buildTypeId=Cockroach_UnitTests>
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/50691#issuecomment-652748014>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA32FQ74BK6N4NARMURYKFDRZPYJVANCNFSM4OJF3BFA>
> .
>


-- 
Oliver Tan | Member of Technical Staff
Cockroach Labs

tbg commented:
Thanks Oliver!

On Thu, Jul 2, 2020 at 6:44 AM craig[bot] <notifications@github.com> wrote:

> Merged #50691 <https://github.com/cockroachdb/cockroach/pull/50691> into
> master.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/50691#event-3505320523>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABGXPZGT7JZMQ5ETLCWYMDTRZQGDPANCNFSM4OJF3BFA>
> .
>

otan commented:
Haha, looks like this didn't work here: https://teamcity.cockroachdb.com/viewLog.html?buildId=2060220&buildTypeId=Cockroach_EssentialCi&tab=buildResultsDiv&branch_Cockroach=50926

no artifacts :\ but it may be indicative of another issue...

otan commented:
nvm, they were just slow to upload!!
